src/internal/connector/graph/concurrency_limiter.go

```diff
@@ -0,0 +1,52 @@
+package graph
+
+import (
+	"net/http"
+	"sync"
+
+	khttp "github.com/microsoft/kiota-http-go"
+)
+
+// ConcurrencyLimiter middleware limits the number of concurrent requests to graph API.
+// Currently we only enforce concurrency limit for exchange
+type ConcurrencyLimiter struct {
```

Comment:
This doesn't need to be exported, unless we plan for external packages to compose their own ConcurrencyLimiter struct, which sounds like something we'd rather avoid.

Reply:
Will do!